commons/zenoh-buffers/src/zslice.rs

```diff
@@ -211,65 +201,9 @@ impl AsRef<[u8]> for ZSlice {
     }
 }
 
-impl Index<usize> for ZSlice {
```

Comment:
Why removing all these impls?

Reply:
(This change is brought by #1289)
Because they are not needed. Rust indexing syntax already includes an auto-deref. I know that some standard collections like `Vec` reimplement `Index` (by just manually doing the deref), but I suspect it may comes from a time where there was no auto-deref.
On the other hand, `bytes::Bytes` doesn't implement `Index`.